src/DataGrid.tsx

```diff
@@ -718,6 +777,11 @@ function DataGrid<R, K extends keyof R, SR>({
       } as unknown as React.CSSProperties}
       ref={gridRef}
       onScroll={handleScroll}
+      onClickCapture={() => {
+        if (selectedPosition.mode === 'SELECT') return;
+        const column = columns[selectedPosition.idx];
+        return column.editor2 !== undefined && column.editor2Options?.commitOnOutsideClick ? onClickCapture : undefined;
```

Comment:
Shouldn't `onClickCapture` be called? Maybe we should do all this at the cell level?

Reply:
I don' t know how it was working during my testing

Reply:
So we commit value on 
- Cell click
- Header click
- Footer click

The only thing left is outside grid click. Not sure how we can handle it at the cell level. Do you mean to always commit the value when we click outside the cell and remove the above logic? The reason I added individual click handlers is to have more control over when the value is committed

Reply:
I have commented `commitOnOutsideClick` for now. Wdyt about adding another api method like `commitChanges`. Users can call `gridRef.current?.commitChanges();` however they like. I think `useClickOutside` is great but it is not flexible

Reply:
I think catching all clicks outside the cell to commit would be great, would clicking on the datagrid border do nothing for example? 🤔
Having a method to commit at any time is also good, we could have both.

>The reason I added individual click handlers is to have more control over when the value is committed

Aren't they all "clicking outside the cell" anyway? Do we handle those in special ways?

Reply:
The issue is the order of the event handlers. Click event on the cell is fired before the outside click. So if we need to add validation and prevent commit for example then it may not be possible with one outside click handler. May be we can have a click handler on the root component to handle commits for header, footer etc 🤔

Reply:
For the validation, do you want to bind inside RDG? I feel for RDG itself, it doesn't need to handle the validation.

So, the user should control the validation state out of the RDG. And whenever any error style / message needs to be display, formatter can directly access outside validation data and render.

By doing so, they can just run validation check after committing and they can decide to eventually dispatch the value or not in `onCommit`.

Reply:
We need some way to prevent default actions like committing value when a different cell is clicked. Editor needs to inform the grid somehow that editor value is invalid so do not commit. Not sure what is the best way to handle it but we need give users some control over grid's internal behavior